File: Archive_Old_Files/refactor.js
const fs = require('fs');
const path = require('path');

const htmlPath = 'index.html';
const outHtmlPath = 'index_clean.html';
const cssDir = 'styles';
const jsDir = 'scripts';
const cssPath = path.join(cssDir, 'main.css');
const jsPath = path.join(jsDir, 'main.js');

if (!fs.existsSync(cssDir)) fs.mkdirSync(cssDir);
if (!fs.existsSync(jsDir)) fs.mkdirSync(jsDir);

let content = fs.readFileSync(htmlPath, 'utf8');

// Extract styles
const styleRegex = /<style[^>]*>([\s\S]*?)<\/style>/gi;
let styles = [];
let match;
while ((match = styleRegex.exec(content)) !== null) {
    styles.push(match[1]);
}
fs.writeFileSync(cssPath, styles.join('\n\n'));

let firstStyle = true;
content = content.replace(styleRegex, (match) => {
    if (firstStyle) {
        firstStyle = false;
        return '<link rel="stylesheet" href="styles/main.css">';
    }
    return '';
});

// Extract inline scripts
const scriptRegex = /<script(?![^>]*src=)[^>]*>([\s\S]*?)<\/script>/gi;
let scripts = [];
while ((match = scriptRegex.exec(content)) !== null) {
    scripts.push(match[1]);
}
fs.writeFileSync(jsPath, scripts.join('\n\n'));

content = content.replace(scriptRegex, '');

// Insert script tag before </body>
content = content.replace(/<\/body>/i, '<script src="scripts/main.js"></script>\n</body>');

fs.writeFileSync(outHtmlPath, content, 'utf8');

console.log(`Extraction complete!`);
console.log(`CSS size: ${fs.statSync(cssPath).size}`);
console.log(`JS size: ${fs.statSync(jsPath).size}`);
console.log(`Clean HTML size: ${fs.statSync(outHtmlPath).size}`);

File: Archive_Old_Files/refactor.py
import os
import re

html_path = 'index.html'
out_html_path = 'index_clean.html'
css_path = 'styles/main.css'
js_path = 'scripts/main.js'

os.makedirs('styles', exist_ok=True)
os.makedirs('scripts', exist_ok=True)

with open(html_path, 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Extract Styles
# Find all <style>...</style> blocks
style_pattern = re.compile(r'<style[^>]*>(.*?)</style>', re.DOTALL | re.IGNORECASE)
styles = style_pattern.findall(content)

with open(css_path, 'w', encoding='utf-8') as f:
    for s in styles:
        f.write(s + '\n')

# Replace the first style block with the link tag, remove the rest
first_style = True
def style_repl(match):
    global first_style
    if first_style:
        first_style = False
        return '<link rel="stylesheet" href="styles/main.css">'
    return ''

content = style_pattern.sub(style_repl, content)

# 2. Extract Scripts (only inline ones, without src=)
# Match <script> tags that DO NOT have a src attribute
script_pattern = re.compile(r'<script(?![^>]*src=)[^>]*>(.*?)</script>', re.DOTALL | re.IGNORECASE)
scripts = script_pattern.findall(content)

with open(js_path, 'w', encoding='utf-8') as f:
    for s in scripts:
        f.write(s + '\n')

# Remove inline scripts
content = script_pattern.sub('', content)

# Insert the main.js script tag before </body>
script_tag = '\n<script src="scripts/main.js"></script>\n</body>'
content = re.sub(r'</body>', script_tag, content, flags=re.IGNORECASE)

with open(out_html_path, 'w', encoding='utf-8') as f:
    f.write(content)

print(f"Extraction complete! CSS size: {os.path.getsize(css_path)}, JS size: {os.path.getsize(js_path)}, Clean HTML size: {os.path.getsize(out_html_path)}")
